activiti-spring-process-extensions/src/test/java/org/activiti/spring/process/variable/DateFormatterProviderTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.spring.process.variable;
+
+import org.junit.Test;
+
+import java.time.DateTimeException;
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.Date;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class DateFormatterProviderTest {
+
+    private DateFormatterProvider provider = new DateFormatterProvider("yyyy-MM-dd[['T'][ ]HH:mm:ss[.SSS'Z']]");
+
+    @Test
+    public void should_returnDate_when_stringRepresentsADate() {
+        //given
+        String dateStr = "1970-01-01";
+
+        //when
+        Date date = provider.toDate(dateStr);
+
+        //then
+        assertThat(date).hasTime(0);
```

Comment:
should not we assert two things? 
1) the date has a date part: 1970-01-01
2) assertThat(date).hasTime(0);

should we add 1) ?

Reply:
Have added check for date parts

Reply:
nice!

Reply:
The only problem is that methods other than `hasTime()` are timezone specific. So the test may fail depending on the location from where you're running your test.